pandas/io/excel/_base.py

```diff
@@ -745,7 +753,26 @@ class ExcelWriter(metaclass=abc.ABCMeta):
     # You also need to register the class with ``register_writer()``.
     # Technically, ExcelWriter implementations don't need to subclass
     # ExcelWriter.
-    def __new__(cls, path, engine=None, **kwargs):
+    def __new__(
+        cls,
+        path: Union[FilePathOrBuffer, ExcelWriter],
+        engine=None,
+        date_format=None,
+        datetime_format=None,
+        mode: str = "w",
+        storage_options: StorageOptions = None,
+        engine_kwargs: Optional[Dict] = None,
+        **kwargs,
+    ):
+        if kwargs:
+            if engine_kwargs is not None:
+                raise ValueError("Cannot use both engine_kwargs and **kwargs")
+            warnings.warn(
+                "Use of **kwargs is deprecated, use engine_kwargs instead.",
+                FutureWarning,
+                stacklevel=2,
+            )
```

Comment:
shouldn't you simply do `engine_kwargs = kwargs` here?

Reply:
Both engine_kwargs and kwargs go entirely unused in `_base.ExcelWriter.__new__` and `_base.ExcelWriter.__init__`; they are only utilized in the subclasses for each engine.